### PR TITLE
close configurable's resources when config is deleted

### PR DESCRIPTION
**Pull request by laurentbh**

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/core/config/YAMLPluginTenantConfigurationHandler.java [hunk 1]
```diff
@@ -69,11 +69,13 @@ public void setDefaultConfigurable(final T defaultConfigurable) {
     @Override
     protected void configure(@Nullable final UUID kbTenantId) {
         final String rawConfiguration = getTenantConfigurationAsString(kbTenantId);
+        T configurable = null;
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/core/config/YAMLPluginTenantConfigurationHandler.java]: Shouldn't `PluginTenantConfigurableConfigurationHandler` be updated as well (it's the key-value version of the YAML one)?

Comment (laurentbh) [reply src/main/java/org/killbill/billing/plugin/core/config/YAMLPluginTenantConfigurationHandler.java]: 936841a

File: src/test/java/org/killbill/billing/plugin/api/notification/TestPluginTenantConfigurableConfigurationHandler.java [hunk 2]
```diff
@@ -81,6 +87,28 @@ public void testConfigurationForTenant() throws Exception {
...
     }
 
+    @Test(groups = "fast")
+    public void testCloseable() throws Exception {
+        final UUID configuredTenant = UUID.randomUUID();
+        final UUID nonConfiguredTenant = UUID.randomUUID();
+
+        // configure tenants
+        mockTenantKvsCloseable(configuredTenant, ImmutableList.<String>of("CloseableTest"), nonConfiguredTenant, ImmutableList.<String>of());
+        closableConfigurationHandler.configure(configuredTenant);
```
Comment (sbrossie) [inline src/test/java/org/killbill/billing/plugin/api/notification/TestPluginTenantConfigurableConfigurationHandler.java]: Is this line really necessary -- the method getConfigurable will call `configure` [here](https://github.com/killbill/killbill-plugin-framework-java/blob/work-for-release-0.21.x/src/main/java/org/killbill/billing/plugin/api/notification/PluginTenantConfigurableConfigurationHandler.java#L66) ?

Comment (laurentbh) [reply src/test/java/org/killbill/billing/plugin/api/notification/TestPluginTenantConfigurableConfigurationHandler.java]: - only for a [new tenant](https://github.com/killbill/killbill-plugin-framework-java/blob/5f1cbae25e8758990cc310e1bab3f0c9035b1d68/src/main/java/org/killbill/billing/plugin/api/notification/PluginTenantConfigurableConfigurationHandler.java#L62) (not in the `configuredTenants` set of the handler)
- `closableConfigurationHandler.configure(configuredTenant);` is the call that would be made when the framework receives a [config update](https://github.com/killbill/killbill-plugin-framework-java/blob/5f1cbae25e8758990cc310e1bab3f0c9035b1d68/src/main/java/org/killbill/billing/plugin/api/notification/PluginConfigurationEventHandler.java#L37)

**Pull request merged**